app/components/sendToken/index.js

```diff
@@ -117,233 +161,558 @@ export const Manual = props => {
+          ]);
+        } else if (
+          BUSDAddressFrom != '0x0000000000000000000000000000000000000000' &&
+          RGPBUSDAddress != '0x0000000000000000000000000000000000000000' &&
+          (RGPBUSDAddress != '0x0000000000000000000000000000000000000000' &&
+            RGPAddressTo != '0x0000000000000000000000000000000000000000')
+        ) {
+          setRoute(
+            `${await selectedToken.symbol} > BUSD > RGP > ${await selectedToToken.symbol}`,
+          );
+          setRouteAddress([
+            selectedToken.address,
+            BSCTestnetTokens.BUSD,
+            BSCTestnetTokens.RGP,
+            selectedToToken.address,
+          ]);
+        } else if (
+          RGPAddressFrom != '0x0000000000000000000000000000000000000000' &&
+          RGPBNBAddress != '0x0000000000000000000000000000000000000000' &&
+          (RGPBNBAddress != '0x0000000000000000000000000000000000000000' &&
+            BNBAddressTo != '0x0000000000000000000000000000000000000000')
+        ) {
+          setRoute(
+            `${await selectedToken.symbol} > RGP > BNB > ${await selectedToToken.symbol}`,
+          );
+          setRouteAddress([
+            selectedToken.address,
+            BSCTestnetTokens.RGP,
+            BSCTestnetTokens.BNB,
+            selectedToToken.address,
+          ]);
+        } else if (
+          BNBAddressFrom != '0x0000000000000000000000000000000000000000' &&
+          RGPBNBAddress != '0x0000000000000000000000000000000000000000' &&
+          (RGPBNBAddress != '0x0000000000000000000000000000000000000000' &&
+            RGPAddressTo != '0x0000000000000000000000000000000000000000')
+        ) {
+          setRoute(
+            `${await selectedToken.symbol} > BNB > RGP > ${await selectedToToken.symbol}`,
+          );
+          setRouteAddress([
+            selectedToken.address,
+            BSCTestnetTokens.BNB,
+            BSCTestnetTokens.RGP,
+            selectedToToken.address,
+          ]);
+        } else if (
+          BUSDAddressFrom != '0x0000000000000000000000000000000000000000' &&
+          BUSDBNBAddress != '0x0000000000000000000000000000000000000000' &&
+          (BUSDBNBAddress != '0x0000000000000000000000000000000000000000' &&
+            BNBAddressTo != '0x0000000000000000000000000000000000000000')
+        ) {
+          setRoute(
+            `${await selectedToken.symbol} > BUSD > BNB > ${await selectedToToken.symbol}`,
+          );
+          setRouteAddress([
+            selectedToken.address,
+            BSCTestnetTokens.BUSD,
+            BSCTestnetTokens.BNB,
+            selectedToToken.address,
+          ]);
+        } else if (
+          BNBAddressFrom != '0x0000000000000000000000000000000000000000' &&
+          BUSDBNBAddress != '0x0000000000000000000000000000000000000000' &&
+          (BUSDBNBAddress != '0x0000000000000000000000000000000000000000' &&
+            BUSDAddressTo != '0x0000000000000000000000000000000000000000')
+        ) {
```

Comment:
Why are these checks made and they seem to be repeated in multiple places. Also kindly make your implementation of this feature dynamic because there are thousands of tokens on the mainnet and it seems your implementation is just focused on the few testnet tokens.

Reply:
it's doesn't focus on a few testnet tokens, those checks are just used to check if a trade is available between the selected tokens and popular tokens (that are likely to have liquidity). it can check thousands of tokens on the mainnet. The only constant thing there are the popular tokens. let me know what you think and if the approach is bad.

Reply:
it will definitely be more dynamic if i make the BNB,RGP,BUSD addresses dynamic based on the chain ID. as for the checks they'll still try to look for trades for thousands of tokens.